fedimint-bitcoind/src/bitcoincore.rs

```diff
@@ -125,7 +144,11 @@ pub fn from_url_to_url_auth(url: &Url) -> anyhow::Result<(String, Auth)> {
             )
         }),
         if url.username().is_empty() {
-            Auth::None
+            let auth = match read_cookie_file() {
+                Ok(cookie) => Some(Auth::UserPass(url.username().to_owned(), cookie)),
+                Err(_) => None,
+            };
+            auth.unwrap_or(Auth::None)
```

Comment:
1. This implementation seems faulty. The cookie file can contain both a user name and a password. [At leas `bitcoincore-rpc` implements it that way.](https://docs.rs/bitcoincore-rpc/0.17.0/src/bitcoincore_rpc/client.rs.html#203-219)
2. Why not just use the `bitcoincore-rpc` implementation in the first place?

Reply:
Oh. I didn't know about `Auth::Cookie`.